test/helpers/envtest.go

```diff
@@ -114,6 +121,93 @@ func NewTestEnvironment() *TestEnvironment {
 	}
 }
 
+func initializeWebhookInEnvironment() {
+	namespacedScopeV1Beta1 := admissionv1beta1.NamespacedScope
+	namespacedScopeV1 := admissionv1.NamespacedScope
+	failedTypeV1Beta1 := admissionv1beta1.Fail
+	failedTypeV1 := admissionv1.Fail
+	equivalentTypeV1Beta1 := admissionv1beta1.Equivalent
+	equivalentTypeV1 := admissionv1.Equivalent
+	noSideEffectsV1Beta1 := admissionv1beta1.SideEffectClassNone
+	noSideEffectsV1 := admissionv1.SideEffectClassNone
+	webhookPathV1 := "/failing"
+
+	env.WebhookInstallOptions = envtest.WebhookInstallOptions{
+		ValidatingWebhooks: []runtime.Object{
+			&admissionv1beta1.ValidatingWebhookConfiguration{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "deployment-validation-webhook-config",
+				},
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "ValidatingWebhookConfiguration",
+					APIVersion: "admissionregistration.k8s.io/v1beta1",
+				},
+				Webhooks: []admissionv1beta1.ValidatingWebhook{
+					{
+						Name: "deployment-validation.kubebuilder.io",
+						Rules: []admissionv1beta1.RuleWithOperations{
+							{
+								Operations: []admissionv1beta1.OperationType{"CREATE", "UPDATE"},
+								Rule: admissionv1beta1.Rule{
+									APIGroups:   []string{"apps"},
+									APIVersions: []string{"v1"},
+									Resources:   []string{"deployments"},
```

Comment:
Is this just a placeholder for testing?

Is the goal to make this parameterized in a way to deploy the webhooks for the CRD resources we are generating? If so, would it be possible to leverage the generated yaml for this somehow? I'd like to see if we can avoid having to replicate content manually for tests where it could get out of sync with the annotations/generated content.

Reply:
It's to enable webhooks in envtest. 
This PR is to fix the issue #2788

Reply:
@Boutheina-Dab my apologies for not being a bit more clear, I was referencing the hardcoding of `apps.v1.deployments` here along with the various other properties of the webhooks (side effects, path, mutating vs non-mutating, etc)

Reply:
@detiber no problem. I got it. Where can I find the generated yaml file of the CRD resources? Sorry, but I'm still new in the codebase of capi.

Reply:
If I may add my two cents to this discussion, 
I think that if our goal is to add webhooks for the core types, we should try to use the webhook configuration yaml from
- /config/webhook
- boostrap/kubeadm/config/webhook
- controlplane/kubeadm/config/webhook

Instead of hardcoding a copy of the webhook definition here, so there is no effort for keeping things in sync.

From a quick look at the controller runtime code, this can be achieved by setting `WebhookInstallOptions.DirectoryPaths` instead of `WebhookInstallOptions. ValidatingWebhooks/MutatingWebhooks` (even do it is not clear to me if calling kustomize is on us or on controller runtime)

Reply:
thanks @fabriziopandini Actually, yes, I was working on that this morning, and I tried to set DirectoryPaths: []string{filepath.Join("config", "webhooks")}, (instead of WebhookInstallOptions. ValidatingWebhooks/MutatingWebhooks). I am testing it. But, didn't include yet:
-  boostrap/kubeadm/config/webhook
- controlplane/kubeadm/config/webhook
(As long as I understand, only manifest yaml file is needed to install webhook, do we need to call kustomization?)

Reply:
That's a good question.
AFAIK, kustomization.yaml defines some patches for the webhooks, like e.g. webhookcainjection_patch.yaml, but TBH, I'm not sure this is required for envtest. 
@vincepri ^^

In the meantime, you can experiment on using the manifest alone

Reply:
It should work using those directly, although it might be worthwhile in the future to run kustomize in memory but let's tackle that when controller-runtime has support for conversion webhooks as well